File: Python/src/subsystems/drive.py

import wpilib
from wpilib.command.subsystem import Subsystem
from wpilib.compressor import Compressor
#from wpilib.solenoid import Solenoid
from wpilib.doublesolenoid import DoubleSolenoid
from commands.tankdrive import TankDrive

from hardware import motor
from values import ids

class Drive(Subsystem):

    def __init__(self):
        super().__init__('Drive')

        # TODO: Change these to another data type which handles any number of motors

        self.LEFT_TRACK = [motor.Motor(ids.LB_motor), motor.Motor(ids.LF_motor)]
        self.RIGHT_TRACK = [motor.Motor(ids.RB_motor), motor.Motor(ids.RF_motor)]

        for MOTOR in self.RIGHT_TRACK:
            MOTOR.setInverted(True)

        #self.pcm0 = Solenoid(ids.PCM_pin0)  
        #self.pcm1 = Solenoid(ids.PCM_pin1)

        #self.pcm = DoubleSolenoid(ids.PCM_pin0, ids.PCM_pin1)
        #self.pcm = None
        #self.pcm.set(False)

    def tank(self, ls, rs):
        for LM in self.LEFT_TRACK:
            LM.set(ls)

        for RM in self.RIGHT_TRACK:
            RM.set(rs)

    def enablePCM(self):
        self.pcm.set(2)
        #self.pcm0.set(True)
        #self.pcm1.set(False)

    def disablePCM(self):
        self.pcm.set(1)

File: Java/src/main/java/org/usfirst/frc/team3966/robot/subsystems/Drive.java
package org.usfirst.frc.team3966.robot.subsystems;

//import edu.wpi.first.wpilibj.command.Command;
import edu.wpi.first.wpilibj.DriverStation;
import org.usfirst.frc.team3966.robot.hardware.PCM;
import org.usfirst.frc.team3966.robot.hardware.DriveMotor;
import edu.wpi.first.wpilibj.command.Subsystem;
import org.usfirst.frc.team3966.robot.commands.TankDrive;
import org.usfirst.frc.team3966.robot.values.IDs;
import org.usfirst.frc.team3966.util.Logger;

/**
 * Drive subsystem, controls left and right motors.
 *
 * Supplies methods for controlling the drive by various functions. The motors
 * objects should not be directly set from any other place.
 */
public class Drive extends Subsystem {
    
    private Logger drive_log;
    
    public Drive() {
        drive_log = new Logger("Drive Subsystem");
    }

    private DriveMotor LB = new DriveMotor(IDs.LB_motor);
    private DriveMotor LF = new DriveMotor(IDs.LF_motor);

    private DriveMotor RB = new DriveMotor(IDs.RB_motor);
    private DriveMotor RF = new DriveMotor(IDs.RF_motor);

    private PCM pcm0 = new PCM(IDs.PCM_node, IDs.PCM_pin0);
    private PCM pcm1 = new PCM(IDs.PCM_node, IDs.PCM_pin1);

    public void initDefaultCommand() {
        RB.setInverted(true);
        RF.setInverted(true);

        pcm0.disable();
        pcm1.enable();

        setDefaultCommand(new TankDrive());
    }

    public void stop() {
        drive(0, 0);
    }
    
    public void drive(double L_speed, double R_speed) {
        LB.set(L_speed);
        LF.set(L_speed);
        
        RB.set(R_speed);
        RF.set(R_speed);
    }
    
    public void enablePCM() {
        pcm0.enable();
        pcm1.disable();
    }

    public void disablePCM() {
        pcm0.disable();
        pcm1.enable();
    }
    
}
